### PR TITLE
[virustotal] add yara results

### PR DESCRIPTION
**Pull request by axelfahy**
### Proposed changes

* Add yara results as  an`Indicator` linked to the `File`, using a `related-to` relationship.
* Minor refactoring based on mypy feedbacks.

Example of yara results: 

```
"crowdsourced_yara_results": [
                {
                    "ruleset_id": "006fbe881a",
                    "rule_name": "XAgent",
                    "ruleset_name": "RussianAPT",
                    "source": "https://github.com/intezer/yara-rules",
                    "author": "Intezer Analyze"
                },
                {
                    "description": "autogenerated rule brought to you by yara-signator",
                    "source": "https://malpedia.caad.fkie.fraunhofer.de/",
                    "author": "Felix Bilstein - yara-signator at cocacoding dot com",
                    "ruleset_name": "win.xagent_auto",
                    "rule_name": "win_xagent_auto",
                    "ruleset_id": "0084eac39c"
                }
            ],
```

In OpenCTI:

The two yara results seen:
Screenshot 2021-09-16 at 16 34 53

The relation with the second one: 
Screenshot 2021-09-16 at 16 35 19

The details of the second one:
Screenshot 2021-09-16 at 16 37 29

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality using different use cases
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: internal-enrichment/virustotal/src/virustotal.py [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
 import requests
-import yaml
 from pycti import OpenCTIConnectorHelper, get_config_variable
-from time import sleep
+from stix2 import Indicator
+import yaml
 
 
 class VirusTotalConnector:
+    """VirusTotal connector."""
```
Comment (marcadamsge) [inline internal-enrichment/virustotal/src/virustotal.py:15]: I think these lines are out of scope of the PR, I guess they appeared during rebase?

File: internal-enrichment/virustotal/src/virustotal.py [hunk 2]
```diff
@@ -118,8 +178,8 @@ def _process_message(self, data):
             )
         return self._process_file(observable)
 
-    # Start the main loop
     def start(self):
+        """Start the main loop."""
```
Comment (marcadamsge) [inline internal-enrichment/virustotal/src/virustotal.py:182]: Same here

**Review by marcadamsge — COMMENTED** [review]
I tested it in local on my machine, it works perfectly 👍

File: internal-enrichment/virustotal/src/virustotal.py (hunk 2 above)
Comment (axelfahy) [reply internal-enrichment/virustotal/src/virustotal.py:182]: The comment was before the method, but according to [PEP257](https://www.python.org/dev/peps/pep-0257/), it's supposed to be below the method name. Moreover, this removes a warning from `mypy`.

File: internal-enrichment/virustotal/src/virustotal.py (hunk 1 above)
Comment (axelfahy) [reply internal-enrichment/virustotal/src/virustotal.py:15]: I have added the docstring, mainly to remove a warning from `mypy` and because the other connectors (e.g. Kaspersky) have a similar docstring below the name of the class. 
I think it is okay to modify it even if this is outside the scope of the PR because in the PR's template, there is a checkbox for refactoring: 

> Where necessary I refactored code to improve the overall quality

Comment (marcadamsge) [reply internal-enrichment/virustotal/src/virustotal.py:15]: A ok, sorry I'm not familiar with the python best practices, fair enough 👍

File: internal-enrichment/virustotal/src/virustotal.py (hunk 2 above)
Comment (marcadamsge) [reply internal-enrichment/virustotal/src/virustotal.py:182]: 👍

File: internal-enrichment/virustotal/src/virustotal.py [hunk 3]
```diff
@@ -28,7 +38,36 @@ def __init__(self):
...
+                    "author": yara.get("author", "No author provided"),
+                    "source": yara.get("source", "No source provided"),
+                    "ruleset_id": ruleset_id,
+                    "ruleset_name": yara.get(
+                        "ruleset_name", "No ruleset name provided"
+                    ),
+                }
+            ),
+            pattern=f"[YARA] {ruleset_id} full rule available on {source}",
+            pattern_type="virustotal-yara",
```
Comment (fdevillard) [inline internal-enrichment/virustotal/src/virustotal.py:67]: It should be `yara` rather than `virustotal-yara` (see: https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_wfiae74706sw)

**Review by fdevillard — APPROVED** [review]

File: internal-enrichment/virustotal/src/virustotal.py (hunk 3 above)
Comment (axelfahy) [reply internal-enrichment/virustotal/src/virustotal.py:67]: Thanks @fdevillard for the link, but the issue is that it's not actually a real yara rule, so if I set `pattern_type=yara`, it will check if the pattern is a yara rule, and it will fail with `ValueError: {'name': 'FunctionalError', 'message': 'Indicator of type yara is not correctly formatted.'}`. That's why I put another name, but maybe there is another solution I'm not seeing?
